modules/http4s/src/smithy4s/http4s/internals/SmithyHttp4sServerEndpoint.scala

```diff
@@ -119,26 +119,22 @@ private[smithy4s] class SmithyHttp4sServerEndpointImpl[F[_], Op[_, _, _, _, _],
       errorTransformation(other).flatMap(F.raiseError)
   }
 
-
-
-  // format: off
-  private val extractInput: (Metadata, Request[F]) ==> I = {
+  private def extractInput(metadata: Metadata, request: Request[F]): F[I] = {
```

Comment:
This isn't exactly what I meant 😅 

This should still be a `val`. The problem with the current shape is that you compile the input decoder on every request, whereas previously it would've happened only once when the `val` was initiated.

I'm thinking of the shape

`private val extractInput: (Metadata, Request[F]) => F[I]`

where the function would be returned as the result of the pattern match.

Reply:
got it. honestly when I rewrote I was like: they're the same let's use a def

but definitely here, a val allows us to do the pattern match ahead of time